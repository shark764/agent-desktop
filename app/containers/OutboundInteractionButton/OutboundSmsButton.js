/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { isPossibleNumber } from 'utils/validator';

import ErrorBoundary from 'components/ErrorBoundary';

import { selectSmsInteractionNumbers } from 'containers/AgentDesktop/selectors';

import OutboundInteractionButton from './index';
import messages from './messages';

export class OutboundSmsButton extends React.Component {
  isEnabled = () =>
    isPossibleNumber(this.props.phoneNumber) &&
    !this.props.smsInteractionNumbers.includes(this.props.phoneNumber);

  getTitle = () => {
    if (!isPossibleNumber(this.props.phoneNumber)) {
      return messages.enterValidPhone;
    } else if (
      this.props.smsInteractionNumbers.includes(this.props.phoneNumber)
    ) {
      return messages.numberInUse;
    } else {
      return this.props.phoneNumber;
    }
  };

  render() {
    return (
      <OutboundInteractionButton
        channelType="sms"
        endpoint={this.props.phoneNumber}
        isEnabled={this.isEnabled()}
        title={this.getTitle()}
      />
    );
  }
}

const mapStateToProps = (state, props) => ({
  smsInteractionNumbers: selectSmsInteractionNumbers(state, props),
});

OutboundSmsButton.propTypes = {
  phoneNumber: PropTypes.string.isRequired,
  smsInteractionNumbers: PropTypes.array.isRequired,
};

export default ErrorBoundary(connect(mapStateToProps)(OutboundSmsButton));
