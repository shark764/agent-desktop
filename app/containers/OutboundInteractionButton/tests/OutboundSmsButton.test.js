/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import { OutboundSmsButton } from '../OutboundSmsButton';

describe('<OutboundSmsButton />', () => {
  it('should be enabled with valid phone number and phone number not in existing sms interactions', () => {
    const rendered = shallow(
      <OutboundSmsButton
        phoneNumber="+15064702497"
        smsInteractionNumbers={[]}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
  it('should be disabled with invalid phone number', () => {
    const rendered = shallow(
      <OutboundSmsButton phoneNumber="invalid" smsInteractionNumbers={[]} />
    );
    expect(rendered).toMatchSnapshot();
  });
  it('should be disabled when phone number is in existing sms interactions', () => {
    const rendered = shallow(
      <OutboundSmsButton
        phoneNumber="+15064701234"
        smsInteractionNumbers={['+15064701234']}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
