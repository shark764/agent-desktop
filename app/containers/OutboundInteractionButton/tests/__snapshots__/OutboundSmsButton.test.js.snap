// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OutboundSmsButton /> should be disabled when phone number is in existing sms interactions 1`] = `
<Connect(ErrorBoundaryComponent)
  channelType="sms"
  endpoint="+15064701234"
  isEnabled={false}
  title={
    Object {
      "defaultMessage": "SMS number is used in existing interaction",
      "id": "app.containers.OutboundInteractionButtons.numberInUse",
    }
  }
/>
`;

exports[`<OutboundSmsButton /> should be disabled with invalid phone number 1`] = `
<Connect(ErrorBoundaryComponent)
  channelType="sms"
  endpoint="invalid"
  isEnabled={false}
  title={
    Object {
      "defaultMessage": "Enter a valid phone number",
      "id": "app.containers.OutboundInteractionButtons.enterValidPhone",
    }
  }
/>
`;

exports[`<OutboundSmsButton /> should be enabled with valid phone number and phone number not in existing sms interactions 1`] = `
<Connect(ErrorBoundaryComponent)
  channelType="sms"
  endpoint="+15064702497"
  isEnabled={true}
  title="+15064702497"
/>
`;
